core/module.h

```diff
@@ -182,9 +171,9 @@ class Module {
         ogates() {}
   virtual ~Module() {}
 
-  pb_error_t Init(const google::protobuf::Any &arg);
+  pb_error_t RunInit(const google::protobuf::Any &arg);
```

Comment:
Renaming this function makes sense. I am not sure whether this function should be private:
- It is meant to be used by external users, whereas private member functions are mostly for internal use.
- There are other public member functions that a derived class does not override. I can move this function to a different public section.
- Google C++ guide does not encourage the usage of friend classes.